### DISCUSSION TITLE
How to correct constantly wrong analog ROI?

### DISCUSSION THREAD

**Discussion by mgaugusch** [jomjol/AI-on-the-edge-device#2949] [General] [answered]
### The Problem

Hi,
I started to use this great project and I'm really excited about it :-)
However, my water meter has a problem with the first digit after comma - it constantly shows ~0.5 m3 too much. 
Is there any option to correct this? 
I looked at AnalogDigitalTransitionStart, but it doesn't seem to be the right one. 

### Version

v15.7.0

### Logfile

```shell
[0d20h06m20s] 2024-02-22T08:02:09 <ERR> [POSTPROC] main: Raw: 0610.5364, Value: , Status: Rate too high - Read: 610.5364 - Pre: 610.4313 - Rate: 0.1051
[0d20h11m20s] 2024-02-22T08:07:09 <ERR> [POSTPROC] main: Raw: 0610.5386, Value: , Status: Rate too high - Read: 610.5386 - Pre: 610.4313 - Rate: 0.1073
[0d20h16m20s] 2024-02-22T08:12:09 <ERR> [POSTPROC] main: Raw: 0610.5690, Value: , Status: Rate too high - Read: 610.5690 - Pre: 610.4313 - Rate: 0.1377
[0d20h31m20s] 2024-02-22T08:27:09 <ERR> [POSTPROC] main: Raw: 0610.5819, Value: , Status: Rate too high - Read: 610.5819 - Pre: 610.4814 - Rate: 0.1005
[0d20h41m20s] 2024-02-22T08:37:09 <ERR> [POSTPROC] main: Raw: 0610.5873, Value: , Status: Rate too high - Read: 610.5873 - Pre: 610.4858 - Rate: 0.1015
[0d20h51m20s] 2024-02-22T08:47:09 <ERR> [POSTPROC] main: Raw: 0610.5873, Value: , Status: Rate too high - Read: 610.5873 - Pre: 610.4873 - Rate: 0.1000
```

### Expected Behavior

Analog meter with constantly wrong offset can be corrected via config parameter.

### Screenshots

![wm](https://github.com/jomjol/AI-on-the-edge-device/assets/24206055/402616ac-6561-4fc9-87ef-f0f85da74bc9)
The value of the "x 0,1" digit is recognized as 5.3, but actually it is 4.88.

**Comment by caco3** [comment]
Looking on your screenshot, the most right analog ROI actually points to something above 5, so 5.3 (actually 5.8..) are correct! How do you think it should be 4.88?

Looking on the meter, I would expect it to read 610.5873 which is exactly what the device also recognises!

**Reply by caco3** [reply, marked as answer]
Or is it that you want to say that your meter is bad and the x.01 pointer is 180 degrees off?
If so, i would get in contact with the energy company and demand a replacement.

**Reply by mgaugusch** [reply]
Exactly, that's the problem :-)
But even if replacement is a mid-term solution (they will take a while for sure), I've just finished this cool project and now it doesn't work :-(
You are probably thinking: It is not much effort to build such an option (CorrectionOffset), but who (apart from me) is ever gonna need it?

**Reply by caco3** [reply]
It is not easy to implement since the ROIs build up on each others. So we would need to add an offset correction for each ROI.

But looking once more on your photo, I still do not get how you get to believe that it is supposed to be 4.88. Can you draw the expected position of the pointers?

If there is simply y consistent error, that would Mo be ok, since one in general is interested on the relative changes, not the absolute number.

What happens if you increase the max rate to > 0.1?

**Comment by SybexX** [comment]
If you think it's so easy, why not do it yourself?
An attempt has been made to solve this problem, but unfortunately it didn't work and caused problems for the majority of users.

**Reply by mgaugusch** [reply]
Do you have any pointers to the attempt that was made? 
I have enough C++ knowledge to do it myself, but I just wanted to check if there is another solution (or if I'm wrong :)).

**Comment by SybexX** [comment]
I think it is still implemented in version 15.6.0 or 15.5.0

The “code/components/jomjol_flowcontroll/ClassFlowPostProcessing.cpp” file is responsible for this

**Reply by caco3** [reply]
> I think it is still implemented in version 15.6.0 or 15.5.0

No, the issue he has here is not releated to any of the recent issues we currently face.
